tests/vmi_multus_test.go

```diff
@@ -1328,28 +1362,30 @@ func configureNodeNetwork(virtClient kubecli.KubevirtClient) {
 	}
 
 	// Helper function returning existing network-config DaemonSet if exists
-	getNetworkConfigDaemonSet := func() *appsv1.DaemonSet {
+	getNetworkConfigDaemonSet := func() (*appsv1.DaemonSet, error) {
 		daemonSet, err := virtClient.AppsV1().DaemonSets(metav1.NamespaceSystem).Get(context.Background(), networkConfigDaemonSet.Name, metav1.GetOptions{})
 		if errors.IsNotFound(err) {
-			return nil
+			return nil, nil
 		}
-		Expect(err).NotTo(HaveOccurred())
-		return daemonSet
+		if err != nil {
+			return nil, err
+		}
+		return daemonSet, nil
 	}
 
 	// If the DaemonSet haven't been created yet, do so
-	runningNetworkConfigDaemonSet := getNetworkConfigDaemonSet()
+	runningNetworkConfigDaemonSet, err := getNetworkConfigDaemonSet()
+	if err != nil {
+		return nil, err
+	}
 	if runningNetworkConfigDaemonSet == nil {
 		_, err := virtClient.AppsV1().DaemonSets(metav1.NamespaceSystem).Create(context.Background(), &networkConfigDaemonSet, metav1.CreateOptions{})
-		Expect(err).NotTo(HaveOccurred())
+		if err != nil {
+			return nil, err
+		}
 	}
 
-	// Make sure that all pods in the Daemon Set finished the configuration
-	nodes := tests.GetAllSchedulableNodes(virtClient)
-	Eventually(func() int {
-		daemonSet := getNetworkConfigDaemonSet()
-		return int(daemonSet.Status.NumberAvailable)
-	}, time.Minute, time.Second).Should(Equal(len(nodes.Items)))
+	return &getNetworkConfigDaemonSet, nil
```

Comment:
I suggest you keep the original loop here and only return the error.
It can be implemented with `wait.PollImmediate` (you should find usage even in this test file).

Reply:
So get rid of eventually and use wait.PollImmediate?

Reply:
Yea, unless you see a problem here.

Reply:
I will try but I had a problem with this one. There was no eventually accepting two args and a map was needed. I had several different thoughts that looked strange. Moving return to a function that and later invoking ready lambda seemed a good solution at a time. I will try your approach but ultimately I can't promise to not land with that one in the end.

Reply:
Have you tried the `wait.PollImmediate` approach?
I am re-reading this again and it really complicates things and is not intuitive.

Expecting `configureNodeNetwork` to return a function that returns a daemonset is too odd.
It does not read right to me.

How about this...
```
nodes := tests.GetAllSchedulableNodes(virtClient)
err := wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
    c, e := getNetworkConfigDaemonSet()
    if e != nil {
        return false, e
    }
    return int(c) == len(nodes.Items), nil
})
if err != nil {
    return err
}
```

Reply:
Yes, I tried, and it did not work. I will evaluate your proposal as my code differed a little bit from yours. If this would fail, I just skip converting this function.
It's your choice in the end, but I tend to disagree that it looks too odd. I see similar constructs there and there all the way across the projects. In the end, is just a matter of convention.